fhirclient/models/fhirsearch.py

```diff
@@ -110,35 +118,64 @@ def include(self, reference_field, reference_model=None, reverse=False):
         self.includes.append((reference_model, reference_field, reverse))
         return self
     
-    def perform(self, server):
+    def perform(self, server) -> 'Bundle':
         """ Construct the search URL and execute it against the given server.
         
         :param server: The server against which to perform the search
         :returns: A Bundle resource
         """
+        # Old method with deprecation warning
+        warnings.warn(
+            "perform() is deprecated and will be removed in a future release. "
+            "Please use perform_iter() instead.",
+            DeprecationWarning,
+        )
+
         if server is None:
             raise Exception("Need a server to perform search")
         
-        from . import bundle
+        from . import bundle as bundle_module
         res = server.request_json(self.construct())
-        bundle = bundle.Bundle(res)
+        bundle = bundle_module.Bundle(res)
         bundle.origin_server = server
         return bundle
```

Comment:
This is not your doing, but this looks like a re-implementation of [Bundle.read_from](https://github.com/LanaNYC/client-py/blob/9967711ef97c4c9c7e6f3553453d8aaa078e1383/fhirclient/models/fhirabstractresource.py#L96).

You might be able to reduce this whole post-deprecation warning method to (it even checks if server is none for you):
```
from .bundle import Bundle
return Bundle.read_from(self.construct(), server)
```